[NNC] add fusion support for nan_to_num

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48975 [NNC] add fusion support for nan_to_num**
* #48976 [NNC] Add fusion support of aten::to
* #48974 [NNC] add support for masked_fill
* #48973 [NNC] Add Support For is_nan

Differential Revision: [D25413167](https://our.internmc.facebook.com/intern/diff/D25413167)

bertmaher reviewed (commented):
it'd be cool to see how nan_to_num performs compared to a fused equivalent of where/isnan (I think that's the equivalent, right?).  It'd be a neat hey-this-what-we-can-do-with-fusion demo.

A few weeks ago @bhosmer and @gchanan and I were talking about what an "explicit fuser" interface would look like, and it'd be neat to see NNC actually being able to be used as a semi-explicit fuser for composite ops like this.

bertmaher commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -911,6 +915,55 @@ Tensor* TensorExprKernel::computeValue(const torch::jit::Value* v) {
+          v,
+          [node](
+              const ExprHandle& input,
+              const ExprHandle& nan,
+              const ExprHandle& pos_inf,
+              const ExprHandle& neg_inf) {
```
I'd be somewhat inclined to name these things like `nan_replacement` (or something).  I was confused seeing `nan` and `pos_inf` being used as default values and thought I'd found a bug when in fact I just demonstrated my lack of reading comprehension :-).

Anyways up to you.  I know the PT API docs use these names so, whatever :)

eellison replied:
yea, agreed that makes it easier to follow

Commit:
Update on "[NNC] add fusion support for nan_to_num"


Differential Revision: [D25413167](https://our.internmc.facebook.com/intern/diff/D25413167)

[ghstack-poisoned]

Commit:
Update on "[NNC] add fusion support for nan_to_num"


Differential Revision: [D25413167](https://our.internmc.facebook.com/intern/diff/D25413167)

[ghstack-poisoned]
